src/lib.rs

```diff
@@ -282,7 +267,7 @@ impl ConvexHull {
             }
         }
 
-        if max_extent <= tolerance {
+        if max_extent == 0.0 {
```

Comment:
Found an related issue: degenerate tetrahedrons have no check for `co{incidence,planarity,linearity}`
```rust
#[test]
fn coincident() {
    for s in [1.0, 10.0, 1000.0] {
        for offset in 0..1000 {
            let mut points = (0..4).map(|_| DVec3::splat(s + offset as f64)).collect::<Vec<_>>();
            // points[2].x += f64::EPSILON;
            let result = ConvexHull::try_new(&points, None);
            assert!(result.is_ok(), "expected ok, got {result:?}")
        }
    }
}
```

Reply:
I am working with the assumption that the algorithm should always return shapes that have positive volume.
I've removed the 4-point shortcut to fix that property for 4-point shapes.

Also found out that several-epsilon-small shapes return negative volume, will think how to fix that.

Reply:
I'll work on making the algorithm determinate first, currently volume calculations fluctuate on the same point cloud

Reply:
I've skipped making the calculation determinate, as I realized that the volume of the compounding tetrahedrons should never be negative, so just clamping it worked fine, the volumes are now positive, even though in theory it can be 0.0, I think the clients might handle that.

I think this solves the last issue of this pr and the determinism issue probably should be revisited in a follow-up